### ISSUE TITLE
[Bug]: Loss of interactive Bash functionality in Pure Mode

### ISSUE THREAD

**Issue by apgrucza** [jetpack-io/devbox#1193]
**Current Behavior (bug)**

When using Bash as a shell and running `devbox shell --pure`, the following is observed:

- The arrow keys input `^[[A` `^[[B` `^[[C` `^[[D` instead of moving the cursor and navigating through the history
- The tab key does not perform command completion

These errors (from an unmodified `.bashrc`) are also shown:

- `bash: shopt: progcomp: invalid shell option name`
- `bash: shopt: hostcomplete: invalid shell option name`
- `bash: complete: command not found`

**Expected Behavior (fix)**

If I add the `bashInteractive` package to my devbox and then type `bash` from inside the pure shell, the problems listed above go away. So, the fix seems to be that Devbox should use the `bashInteractive` package for pure mode, rather than the normal `bash` package.

**Additional context**

Output of `devbox version -v`:

```
Version:     0.5.5
Platform:    linux_amd64
Commit:      891e8de7a8ab7a3fec42fad069326d68cb84603e
Commit Time: 2023-06-19T19:18:00Z
Go Version:  go1.20.5
Launcher:    0.2.0
```

**Comment by Lagoja** [comment]
Arrow keys and tabs definitely look like a bug. Good catch on the bashInteractive package, we can try using that to create the shell to confirm if that fixes things. 

The bashrc errors seem related to your issue #1196. The missing packages make sense (in pure mode, you wouldn't have access to those packages unless you installed them in Devbox), but we should probably not source the `.bashrc` as part of creating the shell

**Comment by apgrucza** [comment]
FYI the errors listed above do go away when using `bashInteractive` as they relate to [Bash Builtin Commands](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html) rather than separate binaries. Issue #1196 covers different errors.

**Comment by mohsenari** [comment]
@apgrucza thanks for reporting this. Will look into this and fix it for next release, if the fix doesn't introduce more complications.
We wanted the pure shell have minimal impact from the system for use cases where absolute reproducibility is crucial. If the only solution is to add a `bashInteractive` package on user's behalf, I'd rather leave that as a suggestion when pure shell is activated.

**Comment by mohsenari** [comment]
This issue should be fixed in 0.5.6
